pkg/webhook/pod/pod.go

```diff
@@ -75,12 +75,12 @@ func (a *podAnnotator) Handle(ctx context.Context, req admission.Request) admiss
 
 // create jsonpatches only to changed caused by the kubemacpool webhook changes
 func patchPodChanges(originalPod, currentPod *corev1.Pod) admission.Response {
-	var kubemapcoolJsonPatches []jsonpatch.Operation
+	kubemapcoolJsonPatches := []jsonpatch.Operation{}
 
 	currentNetworkAnnotation := currentPod.GetAnnotations()[pool_manager.NetworksAnnotation]
 	originalPodNetworkAnnotation := originalPod.GetAnnotations()[pool_manager.NetworksAnnotation]
 	if originalPodNetworkAnnotation != currentNetworkAnnotation {
-		annotationPatch := jsonpatch.NewPatch("add", "/metadata/annotations", map[string]string{pool_manager.NetworksAnnotation: currentNetworkAnnotation})
```

Comment:
I think it override other annotations due to missing `/-` at the patch path section:
```
jsonpatch.NewPatch("add", "/metadata/annotations", ...)
jsonpatch.NewPatch("add", "/metadata/annotations/-", ...)
                                                ^^^
```

Reply:
interesting, will try that!

Reply:
@ormergi I tried using that locally, but am getting:
```
Error from server: error when creating "myvm_2iface_no_ns.yaml": admission webhook "virtualmachine-validator.kubevirt.io" denied the request: json: cannot unmarshal object into Go struct field ObjectMeta.metadata.annotations of type string
```

so I guess we'll do it the kubevirt way